lib/widgets/pokemon_tile_item.dart

```diff
@@ -1,14 +1,23 @@
 import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
-import 'package:pokedex/utils/strings.dart' as str;
+import 'package:pokedex/api/pokedex_api_spec/model/pokemon.dart';
+import 'package:pokedex/utils/constant.dart' as k;
+import 'package:dartx/dartx.dart';
 
 class PokemonTileItem extends StatelessWidget {
-  const PokemonTileItem({super.key});
+  const PokemonTileItem({
+    super.key,
+    required this.pokemon,
+    required this.pokemonImg,
+  });
+
+  final Pokemon pokemon;
+  final String pokemonImg;
```

Comment:
this is not what I meant earlier when I said make this property available 🤦‍♂️, the previous construction was already good it's just that you needed string manipulation to get ready before it's return for widget rendering. 

just have it in between of  `Widget build(BuildContext context) {` and `return GestureDetector(`

that's the importance of asking question when things turned not clarified in your end instead of hastily getting your hands dirty without having better understanding of the problem 😉